Drive/rover_drive_system.hpp

```diff
@@ -79,7 +78,7 @@ class RoverDriveSystem : public sjsu::common::RoverSystem
     int actual_arguments =
         sscanf(response.c_str(), response_body_format,
                &mc_data_.heartbeat_count, &mc_data_.is_operational, &mc_data_.wheel_shift,
-               &mc_data_.drive_mode, &mc_data_.speed, &mc_data_.rotation_angle);
+               &current_drive_mode_, &mc_data_.speed, &mc_data_.rotation_angle);
```

Comment:
major bug!! we need to keep track of current drive mode to compare it to the new mc data drive mode. Directly overwriting the current drive mode makes this impossible